pyproject.toml

```diff
@@ -59,7 +59,6 @@ style = [
 ]
 lint = [
     "qiskit-addon-utils[style]",
-    "pydocstyle==6.3.0",
```

Comment:
I suspect @mrossinek deleted [my comment](https://github.com/Qiskit/qiskit-addon-cutting/blob/c79f9c3d3cfe1f73f4b723e33d1b650eb3f55f29/pyproject.toml#L66) when copying from another repo, but in any case: I believe we can also get rid of `toml` as a lint dependency if we are dropping this.

Max, do you agree?

Reply:
True. I did that in another repository that still had the comment, and CI still passed